src/Command/PelagosImportFilesToDatasetCommand.php

```diff
@@ -0,0 +1,190 @@
+            return 1;
+        }
+
+        $files = $this->dirToList($basePath);
+
+        $fileset = $datasetSubmission->getFileset();
+
+        $deleteFile = null;
+
+        if (!$fileset instanceof Fileset) {
+            $fileset = new Fileset();
+            $datasetSubmission->setFileset($fileset);
+        } else {
+            $deleteFile = $fileset->getProcessedFiles()->first();
+            if ($deleteFile instanceof File) {
+                $physicalFilePath = $deleteFile->getPhysicalFilePath();
+                $pattern = '/^\w\w\.x\d{3}\.\d{3}\:\d{4}\/\w\w\.x\d{3}\.\d{3}\:\d{4}\.dat$/';
+                if (preg_match($pattern, $physicalFilePath) === 1) {
+                    $deleteFileMessage = new DeleteFile($physicalFilePath, true);
+                    $this->messageBus->dispatch($deleteFileMessage);
+                    $fileset->removeFile($deleteFile);
+                } else {
+                    throw new \Exception("Original $physicalFilePath file can not be located in the fileset, QUITTING!");
+                    return 1;
+                }
+            }
+        }
+
+        foreach ($files as $file) {
+            $fileName = $file;
+            $filePath = $basePath . DIRECTORY_SEPARATOR . $file;
+            $fileSize = filesize($filePath);
+
+            if ($fileset->doesFileExist($fileName)) {
+                throw new \Exception("File: $fileName already imported, QUITTING!");
+                return 1;
+            }
+
+            // while ($fileset->doesFileExist($fileName)) {
+                // $fileName = FileNameUtilities::renameFile($fileName);
+            // }
+
+            $newFile = new File();
+            $newFile->setFilePathName(trim($fileName));
+            $newFile->setFileSize($fileSize);
+            $newFile->setUploadedAt(new \DateTime('now'));
+            $newFile->setUploadedBy($systemPerson);
+            $newFile->setPhysicalFilePath($filePath);
+            $newFile->setDescription('Imported by unpack tool.');
+            $newFile->setStatus(File::FILE_IN_QUEUE);
+            $fileset->addFile($newFile);
+            $this->entityManager->persist($newFile);
+        }
+
+        $this->entityManager->flush();
+
+        $datasetSubmissionFilerMessage = new DatasetSubmissionFiler($datasetSubmission->getId());
+        $this->messageBus->dispatch($datasetSubmissionFilerMessage);
+
+        $io->success(sprintf('Files imported for %s.', $udi));
+
+        return 0;
+    }
+
+    /**
+     * This function will take a basepath, and create a list of files and their path.
+     *
+     * @param string $basePath  The base path where the files are located.
+     * @param string $directory A sub directory to look into.
+     * @param array  $files     The list of files to add to.
+     *
+     * @return array The list of files paths in an array.
+     */
+    private function dirToList(string $basePath, string $directory = '', array $files = array()): array
```

Comment:
I'm not sure why this runs so incredibly slow with large nested directories, but a system call to the find command re-directed into a tempfile can be read into a php array incredibly fast. In testing, I was able to read 33k filenames in about 30 ms.  Similarly, the find command produced this much output in a little over a minute.